src/memcached/windowed_cache_impl.go

```diff
@@ -0,0 +1,154 @@
+package memcached
+
+import (
+	"context"
+	"math/rand"
+	"strconv"
+	"sync"
+
+	"github.com/bradfitz/gomemcache/memcache"
+	"github.com/coocood/freecache"
+	pb "github.com/envoyproxy/go-control-plane/envoy/service/ratelimit/v3"
+	"github.com/envoyproxy/ratelimit/src/algorithm"
+	"github.com/envoyproxy/ratelimit/src/config"
+	"github.com/envoyproxy/ratelimit/src/limiter"
+	"github.com/envoyproxy/ratelimit/src/memcached/driver"
+	"github.com/envoyproxy/ratelimit/src/utils"
+	stats "github.com/lyft/gostats"
+	logger "github.com/sirupsen/logrus"
+)
+
+type windowedRateLimitCacheImpl struct {
+	client                     driver.Client
+	timeSource                 utils.TimeSource
+	jitterRand                 *rand.Rand
+	expirationJitterMaxSeconds int64
+	cacheKeyGenerator          utils.CacheKeyGenerator
+	localCache                 *freecache.Cache
+	waitGroup                  sync.WaitGroup
+	nearLimitRatio             float32
+	algorithm                  algorithm.RatelimitAlgorithm
+}
+
+var _ limiter.RateLimitCache = (*windowedRateLimitCacheImpl)(nil)
+
+func (this *windowedRateLimitCacheImpl) DoLimit(
+	ctx context.Context,
+	request *pb.RateLimitRequest,
+	limits []*config.RateLimit) []*pb.RateLimitResponse_DescriptorStatus {
+
+	logger.Debugf("starting cache lookup")
```

Comment:
looks like the code from line 40 up to line 65 is exactly the same as code in fixed_cache impl?

Reply:
yes, most of the code is assigning value to a variable, any concern or suggestion to clean them?